### PR TITLE
Lsn1 exercises

### PR DESCRIPTION
**Pull request by CarlyLanglois**
some outstanding questions:

- does the tutorial video still match the github setup process or should it be removed?

- use a loaner laptop to get screenshots of the intelliJ setup

- do the solutions need to be stored in a new gitlab repo?

### CODE REVIEW

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 1]
```diff
@@ -1,2 +1,152 @@
...
+      Preferences > Appearance & Behavior > System Settings* and uncheck
+      *Reopen last project on startup*. (For Windows users: *File >
+      Settings > Appearance & Behavior > System Settings*.) Close and
+      Reopen IntelliJ.
+
+#. From the “Welcome to IntelliJ” dialog, select *Check out from Version
+   Control > Git*
+   **TODO: Get screenshots of this process.**
+#. Click the button on the lower left corner of the dialog to configure your Github account.
+#. The first time you use GitHub within IntelliJ, you’ll have to
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: The phrasing here seems to be repeating the same idea. How about something like, "To work with a remote repository in IntelliJ, you need to configure the program to access your GitHub account."

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 2]
```diff
@@ -1,2 +1,152 @@
...
+   configure IntelliJ to use your GitHub account. We recommend taking
+   the extra step of authenticating using a token. This takes only one
+   brief extra step, and will prevent you from having to update IntelliJ
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: How about encouraging a security best-practice?  Something like, "...update the ntelliJ settings when you change your GitHub password."

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 3]
```diff
@@ -1,2 +1,152 @@
...
+#. From the URL dropdown options, select your fork of ``java-web-dev-exercises``, 
+   along with an appropriate source destination directory (i.e. a folder where you’ve stored other projects for this class).
+#. When asked “Would you like to create an IDEA project…” select *Yes*, 
+   and then accept all of the default options that are presented.
+
+With that, you’re ready to go!
+
+Here’s a video walking through the last 4 steps above:
+
+.. raw:: html
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: These multiple ``.. raw:: html`` lines can be consolidated into:

.. raw:: html

   <div class="youtube-wrapper">
      <iframe width="560" height="315" src="https://www.youtube.com/embed/OPCaYVXRm_c" frameborder="0" allowfullscreen></iframe>
   </div>

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 4]
```diff
@@ -1,2 +1,152 @@
...
+---------------
+
+ClassNotFoundException
+^^^^^^^^^^^^^^^^^^^^^^
+
+If you experience ``java.lang.ClassNotFoundException`` when trying to
+run code after setting up the project, follow these steps: 
+
+#. Select *File > Close Project*.
+#. From the IntelliJ modal window, click the *X* next to ``java-web-dev-exercises`` in the left-hand pane.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Do the students know what "modal window" means? Is there an IntelliJ workspace overview presented in the videos, or will we need one in the chapter?

Even if the info is in the "Setup" page, a screenshot of the window (with an arrow for more direction) would be helpful.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 5]
```diff
@@ -1,2 +1,152 @@
...
+#. From the same modal window, select *Import Project* from the right-hand pane.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Again, arrows in the previous screenshot to help guide students in this first run-through of IJ.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 6]
```diff
@@ -1,2 +1,152 @@
...
+#. Follow the steps that IntelliJ guides you through, accepting all defaults. When prompted to overwrite IntelliJ settings files, you confirm that you want to do so.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Remove "you" before "confirm".

File: src/chapters/setup-intro-data-types/naming-conventions.rst [hunk 7]
```diff
@@ -0,0 +1,53 @@
+.. _naming-conventions:
+
+Java Naming conventions
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/naming-conventions.rst]: Capitalization fix: "Java Naming Conventions".

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 8]
```diff
@@ -1,2 +1,152 @@
...
+new class for each numbered exercise. You may name the classes whatever you like, but use
+proper :ref:`naming-conventions` and make sure that the file name matches the class name.
+
+Creating a Package and Classes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here is how to create a new package to store these exercises, and how to
+create new classes within this package:
+
+#. Click on the folder ``src`` in the Project pane, then right click and
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: For Mac users, this would be control-click.  For the first couple of "right-click" instructions, we should mention both methods.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 9]
```diff
@@ -1,2 +1,152 @@
...
+      New Package
+
+#. Name the package “exercises”.
+
+   .. figure:: figures/namePackage.png
+      :alt: Name Package
+
+      Name Package
+
+#. Then click on the newly-created folder ``exercises`` and right click
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Word removal suggestion.  How about, "Right-click on the newly created ``exercises`` folder.  Select *New* and then *Java Class*."

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 10]
```diff
@@ -1,2 +1,152 @@
...
+
+5. The new class is now created! You can proceed to write code within
+   it. (Don’t forget to write the ``main`` method!)
+
+   .. figure:: figures/newClassCreated.png
+      :alt: Ready to start
+
+      Ready to start
+
+Exercises
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst:203]: Since these are the first exercises in Java, I'd recommend providing some heavy guidance for exercise 1 and less specific but still relatively complete guidance for exercise 2.  Exercises 3 & 4 are fine as-is, but I'd add a reference back to the Strings method page for exercise 4 "if you need a quick review of the Java String methods."

File: src/chapters/setup-intro-data-types/naming-conventions.rst [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+=======================
+
+Java has some very straightforward naming conventions. These are
+universally used by Java programmers, and differ in some cases from
+conventions commonly used in other languages.
+
+Again, these are conventions. Not following them will not prevent your
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/naming-conventions.rst]: Double nots might not be not easy to not follow.

How about something like, "Ignoring them will not prevent your code from running, as long as you are following Java’s naming rules. However, the conventions are what other Java coders expect to see."

Or we could go with a quote from Pirates of the Caribbean, "The code is more what you’d call ‘guidelines’ than actual rules.”

File: src/chapters/setup-intro-data-types/naming-conventions.rst [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+code from running, as long as you are following Java’s `naming
+rules <http://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.8>`__.
+Java’s identifier naming rules are somewhat hard to parse, so a good
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/naming-conventions.rst]: Should this go in a new paragraph?

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 4 above)
Comment (CarlyLanglois) [reply src/chapters/setup-intro-data-types/exercises.rst]: good call. Im adding screenshots here and changing the language

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 8 above)
Comment (CarlyLanglois) [reply src/chapters/setup-intro-data-types/exercises.rst]: its right click for me on a mac! not for you?

File: src/chapters/setup-intro-data-types/naming-conventions.rst (hunk 11 above)
Comment (CarlyLanglois) [reply src/chapters/setup-intro-data-types/naming-conventions.rst]: changed this to use a little bit of both your suggestions

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 10 above)
Comment (CarlyLanglois) [reply src/chapters/setup-intro-data-types/exercises.rst:203]: good call. will change

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 8 above)
Comment (jimflores5) [reply src/chapters/setup-intro-data-types/exercises.rst]: I use an old PC-designed mouse, so I do have a right-click.  However, for those of us who have single button mice, or who have not changed their settings to enable right-clicking, then control-click is the way to go.

File: src/chapters/setup-intro-data-types/naming-conventions.rst (hunk 12 above)
Comment (CarlyLanglois) [reply src/chapters/setup-intro-data-types/naming-conventions.rst]: yes! i re-worded this based on your other comment from this section

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 5 above)
Comment (CarlyLanglois) [reply src/chapters/setup-intro-data-types/exercises.rst]: i added screenshots but w/o arrows. i think they are pretty straightforward but if not, I can put in arrows

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 13]
```diff
@@ -1,2 +1,210 @@
...
+      Settings > Appearance & Behavior > System Settings*.) Close and
+      Reopen IntelliJ.
+
+#. From the “Welcome to IntelliJ” dialog, select *Check out from Version
+   Control > Git*
+   **TODO: Get screenshots of this process.**
+#. Click the button on the lower left corner of the dialog to configure your Github account.
+#. To work with a remote repository in IntelliJ, you need to configure the program to access 
+   your GitHub account. We recommend taking the extra step of authenticating using a token. 
+   This takes only one brief extra step, and will prevent you from having to update IntelliJ
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: "extra step" is repeated over two sentences.  We can probably rephrase to remove one of the comments.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 14]
```diff
@@ -1,2 +1,210 @@
...
+   settings when you ever change your GitHub password.
+#. From the URL dropdown options, select your fork of ``java-web-dev-exercises``, 
+   along with an appropriate source destination directory (i.e. a folder where you’ve stored other projects for this class).
+#. When asked “Would you like to create an IDEA project…” select *Yes*, 
+   and then accept all of the default options that are presented.
+
+With that, you’re ready to go!
+
+Here’s a video walking through the last 4 steps above:
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: The video is a little outdated, so we should consider replacing it with an updated version or with screenshots.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 15]
```diff
@@ -1,2 +1,210 @@
...
+
+.. raw:: html
+
+   <div class="youtube-wrapper">
+      <iframe width="560" height="315" src="https://www.youtube.com/embed/OPCaYVXRm_c" frameborder="0" allowfullscreen></iframe>
+   </div>
+
+
+
+Troubleshooting
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst:107]: Maybe the Troubleshooting section should be placed farther down the page, say after the students attempt to run the first exercise.

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 5 above)
Comment (jimflores5) [reply src/chapters/setup-intro-data-types/exercises.rst]: The screenshots work as-is.  Let's leave off the arrows.

For consistency with the previous step, how about "startup window" instead of "modal window"?

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 16]
```diff
@@ -1,2 +1,210 @@
...
+      New Package
+
+#. Name the package “exercises”.
+
+   .. figure:: figures/namePackage.png
+      :alt: Name Package
+
+      Name Package
+
+#. Right-click/Control-click on the newly created exercises folder. Select *New* and then *Java Class*.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Code highlighting for ``exercises``.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 17]
```diff
@@ -1,2 +1,210 @@
...
+   .. note:: 
+      You will be asked whether you want to add this file to Git.
+      Press the “Yes” button.
+   
+   .. figure:: figures/addToGit.png
+      :alt: Add class to Git
+
+      Add class to Git
+
+#. The new class is now created! You can proceed to write code within
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Passive to active rephrasing, "You created the new class!"

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 10 above)
Comment (jimflores5) [reply src/chapters/setup-intro-data-types/exercises.rst:203]: I like the updates.  The exercises are much more approachable now.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 18]
```diff
@@ -1,2 +1,210 @@
...
+
+      Ready to start
+
+Exercises
+---------
+
+#. **Input/output**: Write a new “Hello, World” program to prompt the
+   user for their name and greet them by name.
+
+   #. Follow steps 3-5 above to create a new Class inside of your exercises.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: "...to create a new ``HelloWorld`` Class..."

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 19]
```diff
@@ -1,2 +1,210 @@
...
+   #. Add an import statement at the the top of the file to include Scanner:
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Code highlighting on ``Scanner``.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 20]
```diff
@@ -1,2 +1,210 @@
...
+
+      .. sourcecode:: java
+
+         import java.util.Scanner;
+
+   #. Declare a variable of type ``Scanner`` called ``in``:
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: I like the ``input`` route we took with JavaScript.  It seems more descriptive than just ``in``.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 21]
```diff
@@ -1,2 +1,210 @@
...
+
+      .. sourcecode:: java
+
+         Scanner in = new Scanner(System.in);
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: ``Scanner input = new Scanner(System.in)``

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 22]
```diff
@@ -1,2 +1,210 @@
...
+
+      .. sourcecode:: java
+      
+         System.out.println("Hello, what is your name:");
+
+   #. Create a variable to store the user's response using the ``Scanner``'s ``.nextLine()`` method
+
+      .. sourcecode:: java
+
+         String name = in.nextLine();
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: ``input.nextLine();``

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 23]
```diff
@@ -1,2 +1,210 @@
...
+
+   #. Use concatenation to print the greeting:
+
+      .. sourcecode:: java
+
+         System.out.println("Hello " + name);
+   
+   #. Right-click/Control-click the arrow next to your class and run the program.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst:241]: Note to ourselves, in the chapter we should note the 3 ways to run the program:
1) Command line,
2) Right-click name in the sidebar,
3) Menu option.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 24]
```diff
@@ -1,2 +1,210 @@
...
+
+
+#. **Numeric types**: Write a program to calculate the area of a
+   rectangle and print the answer to the console. You should prompt the
+   user for the dimensions. (What data types should the dimensions be?)
+   
+   #. Follow steps 3-5 above to create a new Class inside of your exercises.
+   #. Add an import statement at the top of your file to use ``Scanner``.
+   #. Add a ``Scanner`` object to handle the user's input.
+   #. Add a print line to prompt the user for length of the rectangle.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: "...prompt the user for the length..."

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 25]
```diff
@@ -1,2 +1,210 @@
...
+   #. Define a variable to handle the user's response. Now is the time to know what type the 
+      dimension will be.
+   #. Repeat the previous two steps to ask for and store the rectangle width.
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: "...and store the rectangle's width."

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 26]
```diff
@@ -1,2 +1,210 @@
...
+
+#. **Numeric types**: Write a program to calculate the area of a
+   rectangle and print the answer to the console. You should prompt the
+   user for the dimensions. (What data types should the dimensions be?)
+   
+   #. Follow steps 3-5 above to create a new Class inside of your exercises.
+   #. Add an import statement at the top of your file to use ``Scanner``.
+   #. Add a ``Scanner`` object to handle the user's input.
+   #. Add a print line to prompt the user for length of the rectangle.
+   #. Define a variable to handle the user's response. Now is the time to know what type the
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: Maybe add a note here to replace ``.nextLine()`` with the appropriate ``Scanner`` method.

File: src/chapters/setup-intro-data-types/exercises.rst [hunk 27]
```diff
@@ -1,2 +1,210 @@
...
+      their rectangle is. 
+   #. Run the program to verify your code.
+
+#. **Numeric types**: Write a program that asks a user for the number of
+   miles they have driven and the amount of gas they’ve consumed (in
+   gallons), and print their miles-per-gallon.
+#. **Strings**: The first sentence of *Alice’s Adventures in Wonderland*
+   is below. Store this sentence in a string, and then prompt the user
+   for a term to search for within this string. Print whether or not the
+   search term was found. See if you can make the search
```
Comment (jimflores5) [inline src/chapters/setup-intro-data-types/exercises.rst]: How about rephrasing to make the case-insensitivity mandatory instead of optional.  Something like, "Make the search case-insensitive..."

File: src/chapters/setup-intro-data-types/exercises.rst (hunk 15 above)
Comment (CarlyLanglois) [reply src/chapters/setup-intro-data-types/exercises.rst:107]: i think its placed well here in case they had any issue setting up the project. i added instructions to try running the hello program above this
